packages/api/src/libs/queries.ts

```diff
@@ -102,6 +106,25 @@ export default (appState: Dependencies) => {
     const addresses = Array.from(appState.registeredEmps.values());
     return sumTvl(addresses, currency);
   }
+  async function sumTvm(addresses: string[], currency: CurrencySymbol = "usd") {
+    const tvm = await bluebird.reduce(
```

Comment:
does the `bluebird.reduce` send these calls in parallel? else this could be pretty slow for 100+ calls, no?

Reply:
the bottleneck here would be pulling the data from the database. right now its in memory so its probably fine, but in the case its from a remote db, then there are some tricks we could do to speed it up, but ultimately we are limited by long it takes to get the last tvm value.

Reply:
Agreed. But does the reduce issue the requests all at once? Or does it wait on `n-1` before processing `n`? If that's the case, I think we should use something more like a Promise.all wrapping a map and then reduce the result.

Reply:
it does not request all at once. i will refactor based on promise.all approach